internal/cri/server/status.go

```diff
@@ -61,7 +62,13 @@ func (c *criService) Status(ctx context.Context, r *runtime.StatusRequest) (*run
 		Features:        c.runtimeFeatures,
 	}
 	if r.Verbose {
-		configByt, err := json.Marshal(c.config)
+		type compatConfig struct {
+			criconfig.Config
+			// kubeadm hardcodes config.sandboxImage
+			SandboxImage string `json:"sandboxImage"`
+		}
```

Comment:
The map won't get exported. The bigger problem here is that our config structure should not become an unofficial external interface. This really needs to be documented and fields should be copied into a set of supported fields rather than dumping our config struct.

Reply:
I think the kubeadm implementation followed https://www.hyrumslaw.com/, when it was added back then

Reply:
https://github.com/kubernetes/kubernetes/commit/d12b4d4c6772bc7735c3765d79ecc46c69595088 (k8s 1.27)

Reply:
> The bigger problem here is that our config structure should not become an unofficial external interface. 

In this thread, it was a known shortcoming:
https://github.com/kubernetes/kubeadm/issues/2020#issuecomment-773353657

But it merged anyway, and others retrofitted.
https://github.com/kubernetes/kubernetes/pull/115610#pullrequestreview-1290520466

----

https://github.com/kubernetes/kubernetes/blob/v1.27.0/cmd/kubeadm/app/util/runtime/runtime.go#L180
https://github.com/kubernetes/kubernetes/blob/v1.32.0/cmd/kubeadm/app/util/runtime/runtime.go#L287

Reply:
my point wasn't that we should use the containerd config as an exported structure types, though yes that was done for info fields, my point was this is but a small subset of the interesting image labels for images... A new field, for one of many that would be needed for various images with certain labels...

An array of name value pairs could be provided as kube has interest in the list of pinned images, though they  already get that information from image list with the pinned label and sandbox label. 

https://github.com/containerd/containerd/blob/9b89fb989d91252459cf4c78ac23e2d5b07d2263/internal/cri/labels/labels.go#L26-L33

Reply:
it should be noted that after an upgrade, service or config change we will have some running pods on different pinned sandbox images. We don't force pod restart due to a change in which version we are running.  We've never had an issue to my knowledge, wrt having different pause versions at the same time.

Reply:
The only place we would run into a "problem" is if the pause image desired is not local and we are unable to pull it, due to being disconnected or having a connection issue with the registry.

Reply:
adding config verification, arguably makes us more brittle for connected environments